src/Adapter/Product/Repository/SpecificPriceRepository.php

```diff
@@ -0,0 +1,168 @@
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param SpecificPriceValidator $specificPriceValidator
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        SpecificPriceValidator $specificPriceValidator
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->specificPriceValidator = $specificPriceValidator;
+    }
+
+    /**
+     * @param SpecificPrice $specificPrice
+     * @param int $errorCode
+     *
+     * @return SpecificPriceId
+     *
+     * @throws SpecificPriceConstraintException
+     * @throws CoreException
+     */
+    public function add(SpecificPrice $specificPrice, int $errorCode = 0): SpecificPriceId
+    {
+        $this->specificPriceValidator->validate($specificPrice);
+        $id = $this->addObjectModel($specificPrice, CannotAddSpecificPriceException::class, $errorCode);
+
+        return new SpecificPriceId($id);
+    }
+
+    /**
+     * @param SpecificPriceId $specificPriceId
+     *
+     * @return SpecificPrice
+     *
+     * @throws SpecificPriceNotFoundException
+     */
+    public function get(SpecificPriceId $specificPriceId): SpecificPrice
+    {
+        /** @var SpecificPrice $combination */
+        $combination = $this->getObjectModel(
+            $specificPriceId->getValue(),
+            SpecificPrice::class,
+            SpecificPriceNotFoundException::class
+        );
+
+        return $combination;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param array|null $filters
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function getProductSpecificPrices(ProductId $productId, ?int $limit = null, ?int $offset = null, ?array $filters = []): array
+    {
+        $qb = $this->getSpecificPricesQueryBuilder($productId, $filters)
+            ->select('sp.*')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit)
+        ;
+
+        return $qb->execute()->fetchAll();
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param array|null $filters
+     *
+     * @return int
+     */
+    public function getProductSpecificPricesCount(ProductId $productId, ?array $filters = []): int
+    {
+        $qb = $this->getSpecificPricesQueryBuilder($productId, $filters)
+            ->select('COUNT(sp.id_specific_price) AS total_specific_prices')
+        ;
+
+        return (int) $qb->execute()->fetch()['total_specific_prices'];
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param array|null $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getSpecificPricesQueryBuilder(ProductId $productId, ?array $filters): QueryBuilder
+    {
+        //@todo: filters are not handled.
+        $qb = $this->connection->createQueryBuilder();
```

Comment:
I did not see it the 1st time but you use Doctrine QueryBuilder here? So in this class we mix Doctrine and ObjectModel?

Reply:
Yep like in other repositories we introduced like CombinationRepository
It's the illegitimate child of future and past 😅